File: javascript/test/fib_siblings-spec.js
const expect = require('chai').expect
const fib = require('../src/fib_siblings.js')

describe("fib", function() {
  describe("siblings", function() {
    it ("returns zeroes if given negative numbers", function() {
      expect(fib.siblings(-1)).to.deep.equal([0, 0])
    })

    it ("returns the fibonacci numbers before and after the given number", function() {
      expect(fib.siblings(4)).to.deep.equal([3, 5])
      expect(fib.siblings(7)).to.deep.equal([5, 8])
    })

    it ("returns the previous and next fib numbers, if given a fib number", function() {
      expect(fib.siblings(0)).to.deep.equal([0, 1])
      expect(fib.siblings(1)).to.deep.equal([0, 2])
      expect(fib.siblings(2)).to.deep.equal([1, 3])
      expect(fib.siblings(3)).to.deep.equal([2, 5])
      expect(fib.siblings(5)).to.deep.equal([3, 8])
    })
  })
})

File: ruby/spec/fib_siblings-spec.rb
require './src/fib_siblings.rb'

RSpec.describe Fib, ".siblings" do
  context "given a negative number" do
    it "returns an array of zeroes" do
      expect(Fib.siblings(-1)).to eq([0, 0])
      expect(Fib.siblings(-7)).to eq([0, 0])
    end
  end

  context "given a fibonacci number" do
    it "returns the fibonacci numbers before and after the given number" do
      expect(Fib.siblings(0)).to eq([0, 1])
      expect(Fib.siblings(1)).to eq([0, 2])
      expect(Fib.siblings(2)).to eq([1, 3])
      expect(Fib.siblings(3)).to eq([2, 5])
      expect(Fib.siblings(5)).to eq([3, 8])
    end
  end

  context "given a non-fibonacci number" do
    it "returns the fibonacci numbers before and after the given number" do
      expect(Fib.siblings(4)).to eq([3, 5])
      expect(Fib.siblings(7)).to eq([5, 8])
    end
  end
end
